### PR TITLE
chore(deps): fix dependencies and dependabot config

### PR DESCRIPTION
**Pull request by dreamorosi**
## Description of your changes

This follow up PR to the one that resolved the linked issue tweaks the configuration of Dependant and updates the dependencies versions in all the `package.json`.

The changes to the `dependabot.yml` file remove the redundant workspaces sections from the file as well as add a new dependency group related to AWS CDK, and change the `versioning-strategy` to `increase`. Besides the newly added group, all the other changes are needed for Dependabot to work effectively with the monorepo setup that we are using and that revolves around npm workspaces. For more info see dependabot/dependabot-core#5226.

To align with the new `versioning-strategy` setting, which updates the version of a dependency not only in the `package-lock.json` but also in the respective `package.json` file, I also went and manually updated all the dependencies that were updated by Depdendabot in the past couple weeks.

### Related issues, RFCs

**Issue number:** #1858

## Checklist

- [x] [My changes meet the tenets criteria](https://docs.powertools.aws.dev/lambda/typescript/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [ ] I have made corresponding changes to the *documentation*
- [x] My changes generate *no new warnings*
- [ ] I have *added tests* that prove my change is effective and works
- [x] The PR title follows the [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

***Is it a breaking change?:*** NO

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: packages/parameters/src/dynamodb/DynamoDBProvider.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import {
   DynamoDBClient,
   GetItemCommand,
   paginateQuery,
+  type DynamoDBPaginationConfiguration,
```
Comment (am29d) [inline packages/parameters/src/dynamodb/DynamoDBProvider.ts:6]: This probably slipped unintentional

**Review by am29d — CHANGES_REQUESTED** [review]
Apart from `DynamoDBPaginationConfiguration` we can merge

File: packages/parameters/src/dynamodb/DynamoDBProvider.ts (hunk 1 above)
Comment (dreamorosi) [reply packages/parameters/src/dynamodb/DynamoDBProvider.ts:6]: No, it's intentional. 

In a recent update of the AWS SDK for DynamoDB it appears that the type/shape of the method to paginate (that we [use here](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/packages/parameters/src/dynamodb/DynamoDBProvider.ts#L416)) has changed. Based on the error that TS was giving me I was able to track down the change to [this PR](https://github.com/aws/aws-sdk-js-v3/pull/5288) that was part of v[3.490.0](https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.490.0). 

The PR appears to have added new properties to the DynamoDB client type. On the other hand we were using a more generic type coming from `@aws-sdk/types`. In order to fix the issue I removed the import from `@aws-sdk/types` and used instead the more specific (and correct type).

My bad for not documenting the change more explicitly in the PR.

**Comment by dreamorosi** [comment]
Integration tests for this PR are passing: https://github.com/aws-powertools/powertools-lambda-typescript/actions/runs/7540910276

**Review by am29d — APPROVED** [review]

**Pull request merged**